app/models/user.rb

```diff
@@ -34,14 +35,14 @@ def self.from_omniauth(auth)
     else
       where(provider: auth.provider, uid: auth.uid).first do |user|
         user.password = Devise.friendly_token[0,20]
-        user.full_name = auth.info.name
+        user.first_name = auth.info.name.split.first
+        user.last_name = auth.info.name.split.last
```

Comment:
in case of double surnames? or last names?

Reply:
also, check if it possible to receive such fields separately in JSON from FB
I think its possible